demosplan/DemosPlanCoreBundle/ValueObject/BasicKeycloakUserData.php

```diff
@@ -0,0 +1,150 @@
+ * @method string getLastName()
+ */
+class BasicKeycloakUserData extends ValueObject implements KeycloakUserDataInterface
+{
+    /**
+     * @var array<int, array<int,string>>
+     */
+    protected array $customerRoleRelations = [];
+
+    /**
+     * E-mail-address of the provided user.
+     */
+    protected string $emailAddress = '';
+
+    /**
+     * Unique abbreviation of chosen login name of the provided user.
+     */
+    protected string $userName = '';
+
+    /**
+     * Unique ID of the provided user.
+     */
+    protected string $userId = '';
+
+    /**
+     * Name of the provided organisation.
+     */
+    protected string $organisationName = '';
+
+    /**
+     * Unique identifier of the provided organisation.
+     */
+    protected string $organisationId = '';
+
+    protected string $firstName = '';
+
+    protected string $lastName = '';
+
+    public function fill(ResourceOwnerInterface $resourceOwner): void
+    {
+        $userInformation = $resourceOwner->toArray();
+
+        if (array_key_exists('groups', $userInformation)
+            && is_array($userInformation['groups'])
+        ) {
+            $this->mapCustomerRoles($userInformation['groups']);
+        }
+
+        $this->userId = $userInformation['sub'] ?? '';
+        $this->organisationName = $userInformation['organisationName'] ?? '';
+        $this->organisationId = $userInformation['organisationId'] ?? '';
+        $this->firstName = $userInformation['given_name'] ?? '';
+        $this->lastName = $userInformation['family_name'] ?? '';
+        $this->userName = $userInformation['preferred_username'] ?? ''; // kind of "login" //has to be unique?
+        $this->emailAddress = $userInformation['email'] ?? '';
+
+        $this->lock();
+        $this->checkMandatoryValuesExist();
+    }
+
+    /**
+     * Checks for existing mandatory data.
+     */
+    public function checkMandatoryValuesExist(): void
+    {
+        $missingMandatoryValues = [];
+        if ('' === $this->userId) {
+            $missingMandatoryValues[] = 'userId';
+        }
+
+        if ('' === $this->userName) {
+            $missingMandatoryValues[] = 'userName';
+        }
+
+        if ('' === $this->emailAddress) {
+            $missingMandatoryValues[] = 'emailAddress';
+        }
+
+        if ('' === $this->organisationId) {
+            $missingMandatoryValues[] = 'organisationId';
+        }
+
+        if ('' === $this->firstName && '' === $this->lastName) {
+            $missingMandatoryValues[] = 'name';
+        }
+
+        if ([] === $this->customerRoleRelations) {
+            $missingMandatoryValues[] = 'roles';
+        }
+
+        if ([] !== $missingMandatoryValues) {
+            throw new AuthenticationCredentialsNotFoundException(implode(', ', $missingMandatoryValues).'are missing in requestValues');
+        }
+    }
+
+    /**
+     * @param array<int, string> $groups
+     */
+    private function mapCustomerRoles(mixed $groups): void
```

Comment:
Is this placed correctly in a `BasicKeycloakUserData`? Shouldn't this be placed in some more specific class (or this class be renamed) to something efa/ozg specific?